### PR TITLE
feat(chart): support pull secrets in deployment template

### PR DESCRIPTION
**Pull request by agill17**
Signed-off-by: agill17 <amrit.gill@coveros.com>

Why would someone need this?
When creating custom plugins, the image for that plugin could be stored in a private registry.

### CODE REVIEW

File: charts/velero/values.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ image:
   tag: v1.2.0
   pullPolicy: IfNotPresent
 
+imagePullSecrets: []
```
Comment (carlisia) [inline charts/velero/values.yaml]: Please add documentation to this fields so we know what's for. Thank you!

Comment (agill17) [reply charts/velero/values.yaml]: Done

File: charts/velero/templates/deployment.yaml [hunk 2]
```diff
@@ -139,6 +139,10 @@ spec:
       nodeSelector:
         {{- toYaml . | nindent 8 }}
     {{- end }}
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 6 }}
```
Comment (skriss) [inline charts/velero/templates/deployment.yaml]: curious why this one is `nindent 6` whereas the fields just above and below are `nindent 8`?

Comment (jenting) [reply charts/velero/templates/deployment.yaml]: Is `restic-daemonset` also need this when users choose to deploy restic?

**Comment by carlisia** [comment]
Hi @agill17, are you able to continue with this PR? Seems pretty close to the finish line!

File: charts/velero/templates/deployment.yaml (hunk 2 above)
Comment (agill17) [reply charts/velero/templates/deployment.yaml]: @skriss Nothing big, just personal preference of yaml list indentation
I prefer this;
```
imagePullSecrets:
- name: foo.secret.name
```

over this
```
imagePullSecrets:
  - name: foo.secret.name
```

But not a huge deal. Sorry the delayed response. I have updated `nindent 6` to `nindent 8` to stick with consistency

**Comment by agill17** [comment]
@carlisia This is ready to be re-reviewed.

**Comment by carlisia** [comment]
@agill17 you need to sign the commits.

Need to see why the CI test is failing with this: https://github.com/vmware-tanzu/helm-charts/pull/58/checks?check_run_id=919891052#step:6:252

**Comment by jenting** [comment]
Could you please add the `imagePullSecrets` to restic-daemonset.yaml also? thanks.

**Review by jenting — APPROVED** [review]
wrong click.

Refs to #149

**Comment by jenting** [comment]
#149 merged, close this one.

**Pull request closed without merging**